include/kernel.h

```diff
@@ -549,8 +551,26 @@ enum execution_context_types {
  * produce output.
  *
  * @return N/A
+ *
+ * @deprecated This API is deprecated.  Use k_thread_foreach().
+ */
+__deprecated extern void k_call_stacks_analyze(void);
+
+/**
+ * @brief Iterate over all the threads in the system.
+ *
+ * This routine iterates over all the threads in the system and
+ * calls the user_cb function for each thread.
+ *
+ * @param user_cb Pointer to the user callback function.
+ * @param user_data Pointer to user data.
+ *
+ * @note CONFIG_THREAD_MONITOR must be set for this function to
```

Comment:
The k_thread_foreach() does not produce any output, so this comment is not 100% correct.